app/src/main/java/com/codesoom/assignment/application/ProductService.java

```diff
@@ -0,0 +1,98 @@
+package com.codesoom.assignment.application;
+
+import com.codesoom.assignment.domain.Product;
+import com.codesoom.assignment.domain.ProductRepository;
+import com.codesoom.assignment.errors.ProductNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+/**
+ * 상품을 관리합니다.
+ */
+@Service
+@Transactional
+public class ProductService {
+
+    private final ProductRepository productRepository;
+
+    /**
+     * 상품 서비스 객체를 생성합니다.
+     * @param productRepository 상품 repository
+     */
+    public ProductService(ProductRepository productRepository) {
+        this.productRepository = productRepository;
+    }
+
+    /**
+     * 상품 목록을 반환합니다.
+     * @return 상품 목록
+     */
+    public List<Product> getProducts() {
+        return productRepository.findAll();
+    }
+
+    /**
+     * 주어진 id와 일치하는 상품을 반환합니다.
+     * @param id 상품 id
+     * @return 일치하는 id의 상품
+     */
+    public Product getProduct(Long id) {
+        String format = "아이디 [%d]에 해당하는 상품을 찾지 못했습니다.";
+
+        return productRepository.findById(id)
+                .orElseThrow(() -> new ProductNotFoundException(String.format(format, id)));
+    }
+
+    /**
+     * 상품을 저장하고 저장된 상품을 반환합니다.
+     * @param source 저장할 상품의 source
+     * @return 저장된 상품
+     */
+    public Product saveProduct(Product source) {
+        Product newProduct = Product.createSaveProduct(
+                source.getName(),
+                source.getMaker(),
+                source.getPrice(),
+                source.getImageUrl()
+        );
+
+        return productRepository.save(newProduct);
+    }
+
+    /**
+     * 주어진 id의 상품을 수정하고 수정된 상품을 반환합니다.
+     * @param id 수정할 상품 id
+     * @param source 수정할 상품 source
+     * @return 수정된 상품
+     */
+    public Product updateProduct(Long id, Product source) {
+        String format = "아이디 [%d]에 해당하는 상품을 찾지 못했으므로, 상품을 수정하지 못했습니다.";
+
+        Product product = productRepository.findById(id)
+                .orElseThrow(() -> new ProductNotFoundException(String.format(format, id)));
```

Comment:
예외를 던지는 메소드라면 JavaDoc에 `@throws`도 반드시 넣어주세요. 예외 클래스 타입, 예외가 던져지는 경우 순으로 적어주면 됩니다.
```suggestion
    /**
     * 주어진 id의 상품을 수정하고 수정된 상품을 반환합니다.
     * @param id 수정할 상품 id
     * @param source 수정할 상품 source
     * @return 수정된 상품
     * @throws ProductNotFoundException 상품을 찾지 못한 경우
     */
    public Product updateProduct(Long id, Product source) {
        String format = "아이디 [%d]에 해당하는 상품을 찾지 못했으므로, 상품을 수정하지 못했습니다.";

        Product product = productRepository.findById(id)
                .orElseThrow(() -> new ProductNotFoundException(String.format(format, id)));
```

Reply:
미쳐 생각 못했었는데 꼭 필요했던 작성 케이스였네요.